homeassistant/components/yolink/__init__.py

```diff
@@ -36,6 +37,35 @@
 ]
 
 
+class YoLinkHomeMessageListener(MessageListener):
+    """YoLink home message listener."""
+
+    def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
+        """Init YoLink home message listener."""
+        self._hass = hass
+        self._entry = entry
+
+    def on_message(self, device: YoLinkDevice, msg_data: dict[str, Any]) -> None:
```

Comment:
Will this be called from inside the asyncio event loop?

Reply:
Yes, When mqtt message arrived, on_message method will be call.

Reply:
I've checked the library, and it's using a separate thread for the paho MQTT client loop. So the `on_message` callback will not run inside the event loop. This should be fixed asap, as it will block the event loop used in the core of Home Assistant.

Reply:
Thanks, I have change the mqtt-client, Replace `paho-mqtt` with `asyncio-mqtt` and stop using separate thread for message receive, PR description also changed.